tests/tests_key_size_reissue.yml

```diff
@@ -0,0 +1,64 @@
+---
+- name: Test re-issue certificate if key size changes
+  hosts: all
+  become: false
+  gather_facts: false
+
+  tasks:
+    - name: Issue certificate with key_size 4096
+      ansible.builtin.import_role:
+        name: linux-system-roles.certificate
+      vars:
+        certificate_requests:
+          - name: mycert_key_size
+            dns: www.example.com
+            ca: self-sign
+            key_size: 4096
+
+    - name: Verify each certificate
+      include_tasks: tasks/assert_certificate_parameters.yml
+      loop: "{{ certificates }}"
+      loop_control:
+        loop_var: cert
+      vars:
+        certificates:
+          - path: /etc/pki/tls/certs/mycert_key_size.crt
+            key_path: /etc/pki/tls/private/mycert_key_size.key
+            subject:
+              - name: commonName
+                oid: 2.5.4.3
+                value: www.example.com
+            subject_alt_name:
+              - name: DNS
+                value: www.example.com
+            key_size: 4096
+
+    - name: Issue certificate with key_size 3072
+      ansible.builtin.import_role:
+        name: linux-system-roles.certificate
+      register: result
+      failed_when: not result.changed
```

Comment:
I don't think this works - that is, `import_role` and `include_role` do not report `changed` if the last task reported `changed` - if you really want to test it this way, you'll need to register the result of `certificate_request` to a global role variable that you can export - we do something similar in the firewall role - https://github.com/linux-system-roles/firewall/blob/main/tasks/main.yml#L104 - then use it like https://github.com/linux-system-roles/firewall/blob/2e43f8a1e240d0b89775a61776d2377aa1c91113/tests/tests_zone.yml#L30


* add a `register: __certificate_request_result` to `name: Ensure certificate requests` - I think variables that are registered go into the public namespace, so you should be able to use it from the test
* in the tests, call the role with `include_role` (not `import_role`) and add `public: true` to export the privat

Reply:
hmm - well, the rhel 8.9 test passed - but I'm not even sure if the `failed_when` is being evaluated

Reply:
yes, I forgot to remove the `register` and `failed_when` directives before opening the PR.

I'll fix it along with the CentOS/RHEL 7 fix.

Reply:
fixed